src/amber/validators/params.cr

```diff
@@ -29,6 +32,7 @@ module Amber::Validators
 
     private def call_predicate(params : Amber::Router::Params)
       @value = params[@field]
+      @present = !!params.has_key?(@field)
```

Comment:
what is the `!!` for?

Reply:
`Amber::Router::Params#has_key?` returns `(String | Nil)`, `!!` converts to `Bool`.

See https://github.com/crystal-lang/crystal/blob/c9d1eef8fd/src/hash.cr#L147

Reply:
Oh. I think `Amber::Router::Params#has_key?` should be fixed to check the key exists.

Reply:
I like this and I think this is fine for now. I have opened this issue https://github.com/amberframework/amber/issues/1026 to refactor the `has_key?(key)` to return `Bool` to be consistent with Crystal.

Reply:
You can make the change to the method instead of adding `!!` here https://github.com/amberframework/amber/blob/6c2889bafa57d44d1951a2dec60a519af2fad87a/src/amber/router/params.cr#L45